src/helpers/fetch.ts

```diff
@@ -70,6 +70,17 @@ export async function fetchGetWithinPage<TResult>(
       );
     }
   }, url);
+
+  // Check for automation blocking
+  if (
+    status === 429 ||
+    (result && (result.toLowerCase().includes('block automation') || result.toLowerCase().includes('bot detection')))
+  ) {
+    throw new Error(
+      `Automation detected and blocked by server. Status: ${status}, URL: ${url}. The site is actively blocking automated access. Consider: 1) Using showBrowser:true, 2) Adding longer delays, 3) Using residential proxies, 4) Running at different times of day`,
```

Comment:
I noticed that the errors are different, consider using a single implementation as:

```ts
function assertAutomationNotBlocked(status: number, responseText: string | null, url: string) {
  if (status === 429 || (responseText && /block automation|bot detection/i.test(responseText))) {
    throw new Error(
      `Automation detected and blocked by server. Status: ${status}, URL: ${url}. The site is actively blocking automated access. Consider: 1) Using showBrowser:true, 2) Adding longer delays, 3) Using residential proxies, 4) Running at different times of day`,
    );
  }
}
```

Then call as:
```ts
  if (!ignoreErrors) {
    assertAutomationNotBlocked(status, result, url);
  }
```